### ISSUE TITLE
Add additional step/popup to Calibre-plugin: Choose chapters to download.

### ISSUE THREAD

**Issue by ErskaConti** [JimmXinu/FanFicFare#537]
The idea would be to fetch the provided URL... 
parse it into a list of links, run the filters through them to select the links.
Then request/allow the user to select/remove links to actually download... 

This popup could also show what's already downloaded when entering a URL that already is associated with a book in the library... but I suspect it wouldn't be very welcome in the update-process as that's something I imagine is done in larger batches.

***
Right now I throw a link to fanficfare; and end up downloading a lot of chapters when in actuality I just wanted some of them combined so as to run them through Balaboka for Text To Speech conversion.

example: [Mizuki's Rabbit Range (Drabbles Thread)](https://forums.spacebattles.com/threads/mizukis-rabbit-range-drabbles-thread.380281/threadmarks) over on Spacebattles.com has 333 threadmarks (at time of writing) and multiple different Stories.

As a side "benefit" I feel the additional popup would make the user feel more powerful in the sense of '*Holy shit, look at how much tedium this plugin saves me from*'... and of course a reduction in the load on the webpages.

**Comment by JimmXinu** [comment]
I have to disagree.  I believe that the vast majority of users, the vast majority of the time, want to download (or update) whole stories, not pick and choose chapters.  

Due to the way FFF started life as a CLI tool--and continues to support that use today; the download process, once launched, isn't terribly easy to insert GUI elements into.

You can already choose to limit your download to a range of chapters by putting `[1-5]` for example, after the story URL.  Admittedly, it's not always trivial to determine what the chapter numbers should be ahead of time.  And FFF doesn't remember the range given anywhere--if you later update the story, FFF will pull the rest of the chapters (unless called with a URL with a range again.)

There's also the `ignore_chapter_url_list` feature to explicitly exclude a list of chapters:
```
## You can exclude chapters from a story by listing their chapter URLs
## in ignore_chapter_url_list.  Chapter URLs will be normalized before
## being used, so you can use either longer or normalized chapter
## URLs.
## NOTE: This will cause numWords(word count) to be off for most sites.
## This setting should be used in a specific story's section.
## Example:
[https://forums.spacebattles.com/threads/lengthy-thread-name.100849/]
ignore_chapter_url_list:
 https://forums.spacebattles.com/posts/10157294/
 https://forums.spacebattles.com/posts/10157297/
 https://forums.spacebattles.com/posts/10157299/
 https://forums.spacebattles.com/threads/lengthy-thread-name.100849/#post-10157400
```

You might also consider looking at your use case the other way around--use FFF to download (and later update) the story, then use [EpubSplit](https://www.mobileread.com/forums/showthread.php?t=178799) to make a new book containing the part you want for TTS.

FYI, personally, I've switched from doing off-line TTS to using the [@Voice Aloud Reader](https://play.google.com/store/apps/details?id=com.hyperionics.avar&hl=en_US) app on Android.  I've been pleasantly surprised how well it works.

I'll think about this for a bit longer, but I'm thinking it's not very likely.

**Comment by ErskaConti** [comment]
>I have to disagree. 

Yea, understandable. I too think most users wish to download the whole story... to me it just feels like a natural extra step to have, both to advertise the amount of work being accomplished and to allow for the user to fix edge cases to fit their wants.

>You can already choose to limit your download to a range of chapters by putting [1-5] for example, after the story URL. Admittedly, it's not always trivial to determine what the chapter numbers should be ahead of time. And FFF doesn't remember the range given anywhere--if you later update the story, FFF will pull the rest of the chapters (unless called with a URL with a range again.)

This is the function I actually wanted, tho simply as an GUI-selector... it might be easiest to simply cut out the process of chapter selection completely;

download the page the URL points to twice, the second time with the chapter range selected the first time. (tho the range should be able to be more complex than just a single start-end range)... that way the modification wouldn't intrude on the current process and the only place it would exist would be an extra Calibre-Plugin button/option.

The fact that an Update would download the whole book ignoring this initial setup does not actually matter to me, tho I can see others getting surprised about it if this feature is implemented.

A more complex problem when it comes to splitting after downloading comes in the form of unsorted Threadmarks...

* [[NSFW] Threadmarks for: ENIYA's Story Gacha Hell ft. Truck-sama (Ideas and Snippets)](https://forum.questionablequesting.com/threads/8153/threadmarks) as an example *(QuestionableQuesting NSFW board)* contains 135 threadmarks, and multiple stories with chapters all mixed together.

>FYI, personally, I've switched from doing off-line TTS to using the @Voice Aloud Reader app on Android. I've been pleasantly surprised how well it works.

I too looked into this, and I found it annoyingly cluttered, so I wrote/am writing my own reader that uses the installed TTS engine, and by using IVONA tts with Brian the result is just about the same as on the PC, just a bit longer pauses between sentences. (just gotta polish it up to state where I would feel ok releasing it)

>I'll think about this for a bit longer, but I'm thinking it's not very likely.

Tbh this is a 'minor' annoyance to me, so even just this reply was enough for me to be satisfied.
